app/services/CartServiceImpl.java

```diff
@@ -123,16 +141,27 @@ public int getFrequency(final String cartId) {
 
     @Override
     public Cart createCartWithPactasInfo(final ProductProjection product, final PactasContract contract, final PactasCustomer customer) {
-        final ProductVariant variant = getVariantInContract(product, contract);
-        final Cart cart = sphereClient().execute(CartCreateCommand.of(CartDraft.of(DefaultCurrencyUnits.EUR))).toCompletableFuture().join();
-        Logger.debug("Created new Cart[cartId={}] with Pactas info", cart.getId());
-
-        final AddLineItem action = AddLineItem.of(product.getId(), variant.getId(), 1);
-        final Cart updatedCart = sphereClient().execute(CartUpdateCommand.of(cart, action)).toCompletableFuture().join();
-
-        final Address address = AddressBuilder.of(customer.getCompleteAddress()).build();
-        final Cart cartWithAddress = sphereClient().execute(CartUpdateCommand.of(updatedCart, SetShippingAddress.of(address))).toCompletableFuture().join();
-        return cartWithAddress;
+        return Optional.of(sphereClient().execute(CartCreateCommand.of(CartDraft.of(DefaultCurrencyUnits.EUR))).toCompletableFuture().join())
+                .map((cart) -> {
+                    LOG.debug("Created new Cart[cartId={}] with Pactas info", cart.getId());
+                    final ProductVariant variant = getVariantInContract(product, contract);
+                    final AddLineItem action = AddLineItem.of(product.getId(), variant.getId(), 1);
+                    final Cart updatedCart = sphereClient().execute(CartUpdateCommand.of(cart, action)).toCompletableFuture().join();
+                    final Address address = AddressBuilder.of(customer.getCompleteAddress()).build();
+                    final Cart cartWithAddress = sphereClient().execute(CartUpdateCommand.of(updatedCart, SetShippingAddress.of(address))).toCompletableFuture().join();
+                    return cartWithAddress;
+                }).orElseThrow(() -> new RuntimeException("Unable to create Order"));
```

Comment:
I think the error handling will not work as expected, I suggest to remove the optional block and `).orElseThrow(() -> new RuntimeException("Unable to create Order"));`